docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -167,6 +167,105 @@ no dedicated array type, and any field could contain multiple values. The
 a specific order. See the mapping documentation on <<array, arrays>> for more
 background.
 
+[discrete]
+[[search-fields-nested]]
+==== Handling of nested fields
+
+The `fields` response for <<nested,`nested` fields>> is slightly different from that
+of regular object fields. While leaf values inside regular `object` fields are
+returned as a flat list, values inside `nested` fields are grouped according
+to maintain the independence of each object inside the original nested array.
+For each entry inside a nested field array, values are again returned as a flat list
+unless there are other `nested` fields inside the parent nested object, in which case
+the same procedure is repeated again for the deeper nested fields.
+
+Given the following mapping where `user` is a nested field, after indexing
+the following document and retrieving all fields under the `user` field:
+
+[source,console]
+--------------------------------------------------
+PUT my-index-000001
```

Comment:
It could be clearer to explicitly map the subfields as we do in other places. That way users aren't wondering "what are these new `.keyword` fields"?